### PR TITLE
feat: C2s in own tab (FPLA-1998)

### PR DESCRIPTION
**Pull request by joez-f**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/FPLA-1998

### Change description ###

C2s in separate tab from documents tab. Labels also updated

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: ccd-definition/CaseTypeTab/CareSupervision/8_C2.json [hunk 1]
```diff
@@ -0,0 +1,23 @@
+[
+  {
+    "LiveFrom": "01/01/2017",
+    "CaseTypeID": "CARE_SUPERVISION_EPO",
+    "Channel": "CaseWorker",
+    "TabID": "C2Tab",
+    "TabLabel": "C2",
+    "TabDisplayOrder": 8,
```
Comment (tomaszpowroznik) [inline ccd-definition/CaseTypeTab/CareSupervision/8_C2.json:8]: probably we should reindex other tabs. If 2 tabs have same display order, then real order I guess is unspecified and tabs can change order

File: ccd-definition/CaseTypeTab/CareSupervision/8_C2.json [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  },
+  {
+    "LiveFrom": "01/01/2017",
+    "CaseTypeID": "CARE_SUPERVISION_EPO",
+    "Channel": "CaseWorker",
+    "TabID": "C2Tab",
+    "TabLabel": "C2",
+    "TabDisplayOrder": 8,
+    "CaseFieldID": "c2DocumentBundle",
+    "TabFieldDisplayOrder": 2
```
Comment (tomaszpowroznik) [inline ccd-definition/CaseTypeTab/CareSupervision/8_C2.json:21]: do we need TabShowCondition maybe, if there is no C2 tab will be show

File: service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/UploadC2DocumentsController.java [hunk 3]
```diff
@@ -152,8 +148,7 @@ private void removeDocumentFromData(Map<String, Object> data) {
         ZonedDateTime zonedDateTime = ZonedDateTime.now(ZoneId.of("Europe/London"));
 
         var c2DocumentBundleBuilder = caseData.getTemporaryC2Document().toBuilder()
-            .author(userDetailsService.getUserName())
```
Comment (tomaszpowroznik) [inline service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/UploadC2DocumentsController.java:155]: are we sure about this, recently we refined story that says uploader name should be attached to user. There are 2 conflicting screenshots in story, please double check with bisiness/design

Comment (joez-f) [reply service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/UploadC2DocumentsController.java:155]: checked with design, it was mistake in one of the screenshots and it should be kept it, with content updated from 'Author' -> 'Uploaded by'

File: service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/UploadC2DocumentsAboutToSubmitControllerTest.java [hunk 4]
```diff
@@ -31,7 +31,7 @@
 class UploadC2DocumentsAboutToSubmitControllerTest extends AbstractControllerTest {
     private static final String USER_NAME = "Emma Taylor";
     private static final ZonedDateTime ZONE_DATE_TIME = ZonedDateTime.now(ZoneId.of("Europe/London"));
-    private static final DateTimeFormatter FORMATTER = DateTimeFormatter.ofPattern("h:mma, d MMMM yyyy", Locale.UK);
+    private static final DateTimeFormatter FORMATTER = DateTimeFormatter.ofPattern("d MMMM yyyy, h:mma", Locale.UK);
```
Comment (lewisbirks) [inline service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/UploadC2DocumentsAboutToSubmitControllerTest.java:34]: Could this just be replaced with a call to the DateFormatterHelper?

Comment (lewisbirks) [reply service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/UploadC2DocumentsAboutToSubmitControllerTest.java:34]: Ahh I see it is used to parse from string to localdatetime. There is a method in the helper that converts from string to localdate which seems to be the end result of what you want.

File: ccd-definition/ComplexTypes/CareSupervision/C2DocumentBundle-nonprod.json [hunk 5]
```diff
@@ -0,0 +1,85 @@
+[
```
Comment (lewisbirks) [inline ccd-definition/ComplexTypes/CareSupervision/C2DocumentBundle-nonprod.json:1]: Just double checking but this is just a reordering with a couple of fields renamed?

File: ccd-definition/CaseField/CareSupervision/caseField.json [hunk 6]
```diff
@@ -718,6 +718,14 @@
     "FieldType": "C2ApplicationTypeSelection",
     "SecurityClassification": "Public"
   },
+  {
```
Comment (lewisbirks) [inline ccd-definition/CaseField/CareSupervision/caseField.json:721]: Do you think we should move all these links into their own json file? Might make navigation of the main one slightly easier.

Comment (joez-f) [reply ccd-definition/CaseField/CareSupervision/caseField.json:721]: yeah we should, I could make a ticket for that

File: ccd-definition/ComplexTypes/CareSupervision/C2DocumentBundle-nonprod.json (hunk 5 above)
Comment (joez-f) [reply ccd-definition/ComplexTypes/CareSupervision/C2DocumentBundle-nonprod.json:1]: yep, couple of fields renamed, and tab fields reordered

File: service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/UploadC2DocumentsAboutToSubmitControllerTest.java (hunk 4 above)
Comment (joez-f) [reply service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/UploadC2DocumentsAboutToSubmitControllerTest.java]: Refactored to use Time interface and helper method

File: ccd-definition/CaseField/CareSupervision/caseField.json (hunk 6 above)
Comment (joez-f) [reply ccd-definition/CaseField/CareSupervision/caseField.json:721]: Made a ticket https://tools.hmcts.net/jira/browse/FPLA-2079

File: ccd-definition/CaseTypeTab/CareSupervision/8_C2-nonprod.json [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+  {
+    "LiveFrom": "01/01/2017",
+    "CaseTypeID": "CARE_SUPERVISION_EPO",
+    "Channel": "CaseWorker",
+    "TabID": "C2Tab",
+    "TabLabel": "C2",
+    "TabDisplayOrder": 8,
+    "CaseFieldID": "uploadC2Link",
+    "TabFieldDisplayOrder": 1,
+    "FieldShowCondition": "c2DocumentBundle!=\"\"",
```
Comment (tomaszpowroznik) [inline ccd-definition/CaseTypeTab/CareSupervision/8_C2-nonprod.json:11]: dont think that we need FielShowCondition, tab will be hidden anyway

File: ccd-definition/CaseTypeTab/CareSupervision/7_Documents.json [hunk 8]
```diff
@@ -148,6 +138,6 @@
     "TabLabel": "Documents",
     "TabDisplayOrder": 7,
     "CaseFieldID": "scannedDocuments",
-    "TabFieldDisplayOrder": 15
+    "TabFieldDisplayOrder": 14
```
Comment (tomaszpowroznik) [inline ccd-definition/CaseTypeTab/CareSupervision/7_Documents.json:141]: i would keep it 15, otherwise order is unpredictable on prod

**Pull request merged**